MSMAll/scripts/SingleSubjectConcat.sh

```diff
@@ -163,31 +141,41 @@ for ((index = 0; index < ${#fMRINamesArray[@]}; ++index)) ; do
 	FrameOutput=${ResultsFolder}/${fMRIName}${fMRIProcSTRING}_${FrameString}${OutputProcSTRING}.dtseries.nii
 
 	# pick frames
-	${Caret7_Command} -cifti-merge ${FrameDenseTCS} -cifti ${FullDenseTCS} -column ${StartFrame} -up-to ${EndFrame}
+	if [[ "$EndFrame" == "" ]]; then
+		if ((StartFrame==1)); then
+			# override the FrameDenseTCS with the full dense timeseries
+			FrameDenseTCS=${FullDenseTCS}
+			# mark temp files for mean, and timeseries after the above process
+			tempfiles_add ${FrameMean} ${FrameOutput}
+		else
+			${Caret7_Command} -cifti-merge ${FrameDenseTCS} -cifti ${FullDenseTCS} -column ${StartFrame} -up-to ${runFrames[index]}
+			# mark temp files for mean, selected range and timeseries after the above process
+			tempfiles_add ${FrameMean} ${FrameDenseTCS} ${FrameOutput}
+		fi
+	else
+		${Caret7_Command} -cifti-merge ${FrameDenseTCS} -cifti ${FullDenseTCS} -column ${StartFrame} -up-to ${EndFrame}
+		# mark temp files for mean, selected range and timeseries after the above process
+		tempfiles_add ${FrameMean} ${FrameDenseTCS} ${FrameOutput}
+	fi
 
 	# mean file
 	${Caret7_Command} -cifti-reduce ${FrameDenseTCS} MEAN ${FrameMean}
-	MATHDemean=" - Mean"
 	VarDemean="-var Mean ${FrameMean} -select 1 1 -repeat"
```

Comment:
As long as these are going to remain unconditional, I wouldn't put the `-var` argument sequences in variables, either.